app/services/page_updater.rb

```diff
@@ -81,9 +81,10 @@ def refresh?
 
   def update_page
     return unless @params[:page]
+    @page.slug = nil
```

Comment:
That's a great point. Autosave does not play nicely with this at all. Let's scrap this line and it'll use the first-made slug.